File: spec/files/plural/Strings.m

#import "Strings.h"

NSString* LocalizedSingularString(SingularString singularString) {
    switch (singularString) {
        case SingularString_first_key:
            return NSLocalizedString(@"First key", @"");
        case SingularString_key_with_format:
            return NSLocalizedString(@"Key with format %ld", @"");
        case SingularString_key_without_description:
            return NSLocalizedString(@"Key without description", @"");
        default:
            return @"";
    }
}

NSString* NSStringFromPluralString(PluralString pluralString) {
    switch (pluralString) {
        case PluralString_downloads_remaining:
            return NSLocalizedString(@"%lu downloads remaining", @"");
        case PluralString_souls_days:
            return NSLocalizedString(@"%lu souls, %lu days", @"");
        default:
            return @"";
    }
}

NSString* LocalizedPluralString(PluralString pluralString, ...) {
    va_list args;
    va_start(args, pluralString);
    NSString *string = [[NSString alloc] initWithFormat:NSStringFromPluralString(pluralString) locale:[NSLocale currentLocale] arguments:args];
    va_end(args);
    return string;
}

File: spec/files/plural/Strings.swift
import Foundation

struct Strings {

    enum Singular: String {
        case first_key = "First key"
        case key_with_format = "Key with format %ld"
        case key_without_description = "Key without description"

        var localizedString: String {
            return NSLocalizedString(rawValue, comment: "")
        }
    }

    enum Plural: String {
        case downloads_remaining = "%lu downloads remaining"
        case souls_days = "%lu souls, %lu days"

        func localizedString(args: CVarArgType...) -> String {
            let localized = NSLocalizedString(rawValue, comment: "")
            return String(format: localized, locale: NSLocale.currentLocale(), arguments: args)
        }
    }
}
